pages/_document.js

```diff
@@ -12,9 +12,48 @@ export default class MyDocument extends Document {
   render() {
     return (
       <html>
-        <Head />
+        <Head>
+          <meta name='viewport' content='width=device-width, initial-scale=1' />
+
+          <meta name='theme-color' content='#ffffff' />
+          <link rel='apple-touch-icon' sizes='180x180' href='/static/favicons/apple-icon-180x180.png' />
+          <link rel='icon' type='image/png' sizes='16x16' href='/static/favicons/favicon-16x16.png' />
+          <link rel='icon' type='image/png' sizes='32x32' href='/static/favicons/favicon-32x32.png' />
+          <link rel='manifest' href='/static/favicons/manifest.json' />
+          <link rel='mask-icon' href='/static/favicons/safari-pinned-tab.svg' color='#5bbad5' />
+
+          <link rel='preload' href='/static/fonts/lato-regular-webfont.woff2' as='font' type='font/woff2' crossOrigin='' />
+        </Head>
         <body>
           <Main />
+          <script dangerouslySetInnerHTML={{
+            __html: `
+              (function () {
+                if ('fonts' in document) {
```

Comment:
Beware that it excludes [all IEs so far](https://caniuse.com/#feat=font-loading), that's why I still recommend [fontfaceobserver](https://github.com/bramstein/fontfaceobserver) but YMMV.

Reply:
I know, but fontfaceobserver is too heavy. IE will just have to deal with FOIT in the first render.